backend/danswer/background/celery/celery_utils.py

```diff
@@ -84,25 +103,36 @@ def should_prune_cc_pair(
     return time_since_last_pruning.total_seconds() >= connector.prune_freq
 
 
+def document_batch_to_ids(doc_batch: list[Document]) -> set[str]:
+    return {doc.id for doc in doc_batch}
+
+
 def extract_ids_from_runnable_connector(runnable_connector: BaseConnector) -> set[str]:
     """
     If the PruneConnector hasnt been implemented for the given connector, just pull
     all docs using the load_from_state and grab out the IDs
     """
     all_connector_doc_ids: set[str] = set()
+
+    doc_batch_generator = None
     if isinstance(runnable_connector, IdConnector):
         all_connector_doc_ids = runnable_connector.retrieve_all_source_ids()
     elif isinstance(runnable_connector, LoadConnector):
         doc_batch_generator = runnable_connector.load_from_state()
-        for doc_batch in doc_batch_generator:
-            all_connector_doc_ids.update(doc.id for doc in doc_batch)
     elif isinstance(runnable_connector, PollConnector):
         start = datetime(1970, 1, 1, tzinfo=timezone.utc).timestamp()
         end = datetime.now(timezone.utc).timestamp()
         doc_batch_generator = runnable_connector.poll_source(start=start, end=end)
-        for doc_batch in doc_batch_generator:
-            all_connector_doc_ids.update(doc.id for doc in doc_batch)
     else:
         raise RuntimeError("Pruning job could not find a valid runnable_connector.")
 
+    if doc_batch_generator:
+        doc_generation_func = document_batch_to_ids
+        if MAX_PRUNING_DOCUMENT_RETRIEVAL_PER_MINUTE:
+            doc_generation_func = rate_limit_builder(
```

Comment:
nit: should be more accurately called something like `doc_batch_processing_func`